docs/developer_guide/style_guidelines_and_tests.md

```diff
@@ -1,35 +1,94 @@
-# Style Guidelines
+# Coding Style Guidelines
 
-We expect contributions to mimic existing patterns in the codebase and demonstrate good practices: the code should be concise, readable, PEP8-compliant, and conforming to 80 character line length limit.
+We expect contributions to mimic existing patterns in the codebase and demonstrate good practices: the code should be
+concise, readable, [PEP8-compliant](https://peps.python.org/pep-0008/), and limit each line to 120 characters.
+
+See [codebase structure](../codebase_structure) for guidelines on where new modules should be added.
+
+## pre-commit.ci
+
+The Ludwig repository integrates with [pre-commit.ci](https://pre-commit.ci/), which enforces basic code style
+guidelines and automatically fixes minor style issues by adding a commit to pull requests. So, check the results of
+pre-commit.ci after creating a new pull request. There may be automatic fixes to pull, or issues which require manual
+editing to fix.
+
+## Docstrings
+
+All new files, classes, and methods should have a docstring. Parameters are documented using
+[Sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#python-signatures)-style signatures.
```

Comment:
I think Google style is more readable in-code (makes sense at google, nobody generates docs everyone uses code search), whereas Sphinx is more formal-looking and easier to parse to generate docs with links to types. 

I don't have a strong preference myself (I weakly lean toward Google Style), but the docstrings in `ludwig/api.py` look like Sphinx so I think we should go with that for now.

Reply:
I agree with you that we want to prescriptive of the style in general :)
I was suggesting not to be perscriptive now because of the stage where we are at in our decision making about how to generate parts of the docs from docstrings.

If we were to make a decision right now, given the manual codegen script we have, we should suggest to use restructuredtext-like syntax for the parameters and markdown syntax in the textual descriptions, which isn't great, and something we likely want to change in the near future.

In order to make a forward looking decision on the other hand, we would need first to make a decision regarding the docs. If we want to replace mkdocs with Sphinx, we can and we would suggest to use restructuredtext (but we will have to change all our .md files to restructuredtext which is quite the undertaking). If we decided to use docusaurus, maybe the choice is different.
As I believe it's a consequential choice that needs to be investigated deeply. I suggested we may want to avoid being prescriptive right now and start being prescriptive after this important decision is made.

Reply:
Fair points!  I'll hold off on prescriptive specifics that reference a particular tool for now.